### ISSUE TITLE
Segments Returning 0 or null Values

### ISSUE THREAD

**Issue by ddeppner** [googleads/google-ads-php#29]
Issuing a search query like this:

select segments.day_of_week, segments.hour, metrics.clicks, metrics.conversions, metrics.conversions_value from customer where segments.date BETWEEN "2018-02-23" AND "2019-02-23"

The segmentation appears to be working correctly, because I get the proper number of result rows (24 hours a day * 7 days a week = 168).

When I try to get data from the response row using getDayOfWeek() or getHour() I get invalid data on the segments, but valid data on the metrics for each row, like this:

[
      "segmentsDayOfWeek" => 0
      "segmentsHour" => null
      "metricsClicks" => 1948
      "metricsConversions" => 55.0
      "metricsConversionsValue" => 3359.66
]

day_of_week is always 0.  hour is always null.

I'm uncertain if this is an API issue (are segments not implemented fully yet?) or a PHP library issue.

**Comment by Raibaz** [comment]
Hi,

How are you getting that data from the response row?

I just tried running your query and then iterating through the rows in the response like this:

    printf("%s %s %s %s %s",
        $googleAdsRow->getSegments()->getDayOfWeek(), 
        $googleAdsRow->getSegments()->getHour()->getValue(), 
        $googleAdsRow->getMetrics()->getClicks()->getValue(), 
        $googleAdsRow->getMetrics()->getConversions()->getValue(), 
        PHP_EOL
    );

And I'm getting all the values I'd expect in the segments, with no nulls or zeroes.

Could it be you are not looking in the right properties of your row objects?

**Comment by ddeppner** [comment]
@Raibaz,

Okay, I figured it out.  The issue was due to the 0.6 version of the google-ads-api client library being installed.  Once I updated to 0.7, and applied the patched GoogleAdsGapicClientTrait.php file, it worked perfectly.  Looks like the segment data got moved around in 0.7.

When I Googled this, I found other references to people having this issue, and I assumed it was a legit bug.  Apparently they need to update their library version as well.

Thanks for your response.

David
